File: sdk/java/sgraal-java/src/main/java/com/sgraal/MemoryEntry.java
package com.sgraal;

public class MemoryEntry {
    public String id;
    public String content;
    public String type;
    public int timestampAgeDays;
    public double sourceTrust;
    public double sourceConflict;
    public int downstreamCount;

    public MemoryEntry(String id, String content, String type, int timestampAgeDays,
                       double sourceTrust, double sourceConflict, int downstreamCount) {
        this.id = id; this.content = content; this.type = type;
        this.timestampAgeDays = timestampAgeDays; this.sourceTrust = sourceTrust;
        this.sourceConflict = sourceConflict; this.downstreamCount = downstreamCount;
    }
}

File: sdk/dotnet/Sgraal/Models/MemoryEntry.cs
namespace Sgraal.Models;
public class MemoryEntry
{
    public string Id { get; set; } = "";
    public string Content { get; set; } = "";
    public string Type { get; set; } = "semantic";
    public double TimestampAgeDays { get; set; }
    public double SourceTrust { get; set; } = 0.9;
    public double SourceConflict { get; set; } = 0.05;
    public int DownstreamCount { get; set; } = 1;
}
